pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -458,7 +471,7 @@ func targetPodMatchesAffinityOfPod(pod, targetPod *v1.Pod) bool {
 		glog.Errorf("error in getting affinity properties of Pod %v", pod.Name)
 		return false
 	}
-	return podMatchesAffinityTermProperties(targetPod, affinityProperties)
+	return podMatchesAllAffinityTermProperties(targetPod, affinityProperties)
```

Comment:
Did not get time to go through all doc and comments :( 

But according to the comments of `PodAntiAffinity`, it said 

> 2309     // If the anti-affinity requirements specified by this field are not met at
2310     // scheduling time, the pod will not be scheduled onto the node.
2311     // If the anti-affinity requirements specified by this field cease to be met
2312     // at some point during pod execution (e.g. due to a pod label update), the
2313     // system may or may not try to eventually evict the pod from its node.
2314     // When there are multiple elements, the lists of nodes corresponding to each
2315     // podAffinityTerm are intersected, i.e. all terms must be satisfied.
2316     // +optional

And how do we handle Affinity that Node did not have such label? I'm thinking that a `bool` (Yes or No) maybe not enough :)

Reply:
"Node" is a typo I guess... I suppose you meant "pod".

If there is no pod has labels matching the terms, function `podMatches{All|Any}AffinityTermProperties (targetPod, affinityProperties)` both returns false.

This is enough I think, because:
> There are 2 phases to verify symmetry of PodAntiAffinity:
> - verify labels of incoming pod not violates {anti}affinity of existing pods
> - verify {anti}affinity of incoming pod not violates labels of existing pods 

In other words, we need to evaluate combination of each phase: (can do early break for sure)
- if it's Affinity case
    - only and only if `true` + `true` means fit
    - other combinations means no fit
- if it's AntiAffinity case
    - only and only if `false` + `false` means fit
    - other combinations means no fit